components/chart/src/tooltip/chart-tooltip-container.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Component,
+  ChangeDetectorRef,
+  ElementRef,
+  EmbeddedViewRef,
+  NgZone,
+  OnDestroy,
+  ViewChild,
+} from '@angular/core';
+import {
+  AnimationEvent,
+  trigger,
+  state,
+  style,
+  transition,
+  animate,
+} from '@angular/animations';
+
+import { DT_CHART_TOOLTIP_FADE_TIME } from './chart-tooltip-config';
+import {
+  mixinNotifyDomExit,
+  HasNgZone,
+  CanNotifyOnExit,
+} from '@dynatrace/barista-components/core';
+import { Subject } from 'rxjs';
+import { CdkPortalOutlet, TemplatePortal } from '@angular/cdk/portal';
+
+// Boilerplate for applying mixins to DtChartTooltipContainer.
+export class DtChartTooltipContainerBase implements HasNgZone {
+  constructor(public _ngZone: NgZone) {}
+}
+export const _DtChartTooltipContainerMixin = mixinNotifyDomExit(
+  DtChartTooltipContainerBase,
+);
+
+@Component({
+  selector: 'dt-chart-tooltip-container',
+  template: '<ng-template cdkPortalOutlet></ng-template>',
+  styleUrls: ['chart-tooltip-container.scss'],
+  host: {
+    class: 'dt-chart-tooltip-container',
+    '[@fade]': '_animationState',
+    '(@fade.done)': '_animationDone($event)',
+  },
+  animations: [
+    trigger('fade', [
+      state('void', style({ opacity: 0 })),
+      transition(
+        ':enter',
+        animate(`${DT_CHART_TOOLTIP_FADE_TIME}ms ease-in-out`),
+      ),
+      transition(
+        ':leave',
+        animate(`${DT_CHART_TOOLTIP_FADE_TIME}ms ease-in-out`),
+      ),
+    ]),
+  ],
+})
+export class DtChartTooltipContainer extends _DtChartTooltipContainerMixin
+  implements OnDestroy, CanNotifyOnExit {
+  private _destroyed = false;
+
+  @ViewChild(CdkPortalOutlet, { static: true }) _portalOutlet: CdkPortalOutlet;
+
+  afterClosed = new Subject<void>();
+
+  /** @internal The current state of the animation. */
+  _animationState: 'void' | 'enter' = 'void';
+
+  constructor(
+    public _ngZone: NgZone,
```

Comment:
Why do you need the is there any reason why the zone is public? you only need it to pass it to the super class.
Why do we need the dom exit notify in general?
```suggestion
    ngZone: NgZone,
```